azext_iot/tests/iothub/configurations/test_iot_config_unit.py

```diff
@@ -902,10 +903,14 @@ def test_config_list_invalid_args(self, fixture_cmd, top):
 
     def test_config_list_error(self, fixture_cmd, service_client_generic_errors):
         service_client_generic_errors.assert_all_requests_are_fired = False
-        with pytest.raises(CLIError):
+        with pytest.raises((CLIError, ClientRequestError)) as e:
```

Comment:
The error tests need to assert what the application logic will throw rather than the service client unless the command logic does not catch the error, why would a new responses version change that?

Reply:
The ClientRequestError is not caught by the command logic and this seems to align with my testing with a broken IoT Hub.

Reply:
The client is supposed to throw a `CloudError` if the response is not 200

![image](https://user-images.githubusercontent.com/28658112/217665013-9f8c30d3-2ce1-4d8c-91d4-378b9400f6bc.png)

And that is what the command handler is expecting

![image](https://user-images.githubusercontent.com/28658112/217665567-3f0f8135-586a-4990-ba18-085edc3cfdf7.png)

Reply:
The line above - aka the 273 response, now throws the error for a maxretry (500) service error

So the code never gets to checking the response.status_code and the error is not caught by our handle_service_exception

Reply:
now implying in responses 0.22.0 where it works as expected in 0.20.0